presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestMultiJoinNodeBuilder.java

```diff
@@ -0,0 +1,332 @@
+        Symbol b1 = planBuilder.symbol("B1", BIGINT);
+        Symbol c1 = planBuilder.symbol("C1", BIGINT);
+        Symbol d1 = planBuilder.symbol("D1", BIGINT);
+        Symbol d2 = planBuilder.symbol("D2", BIGINT);
+        Symbol e1 = planBuilder.symbol("E1", BIGINT);
+        Symbol e2 = planBuilder.symbol("E2", BIGINT);
+        ValuesNode valuesA = planBuilder.values(a1);
+        ValuesNode valuesB = planBuilder.values(b1);
+        ValuesNode valuesC = planBuilder.values(c1);
+        ValuesNode valuesD = planBuilder.values(d1, d2);
+        ValuesNode valuesE = planBuilder.values(e1, e2);
+        JoinNode join1 = planBuilder.join(
+                INNER,
+                valuesA,
+                valuesB,
+                ImmutableList.of(new JoinNode.EquiJoinClause(a1, b1)),
+                ImmutableList.of(a1, b1),
+                Optional.empty());
+        JoinNode join2 = planBuilder.join(
+                INNER,
+                valuesD,
+                valuesE,
+                ImmutableList.of(
+                        new JoinNode.EquiJoinClause(d1, e1),
+                        new JoinNode.EquiJoinClause(d2, e2)),
+                ImmutableList.of(
+                        d1,
+                        d2,
+                        e1,
+                        e2),
+                Optional.empty());
+        JoinNode joinNode = planBuilder.join(
+                INNER,
+                planBuilder.join(
+                        INNER,
+                        join1,
+                        valuesC,
+                        ImmutableList.of(new JoinNode.EquiJoinClause(a1, c1)),
+                        ImmutableList.of(a1, b1, c1),
+                        Optional.empty()),
+                join2,
+                ImmutableList.of(new JoinNode.EquiJoinClause(b1, e1)),
+                ImmutableList.of(
+                        a1,
+                        b1,
+                        c1,
+                        d1,
+                        d2,
+                        e1,
+                        e2),
+                Optional.empty());
+        MultiJoinNode expected = new MultiJoinNode(
+                ImmutableList.of(join1, join2, valuesC),
+                and(
+                        new ComparisonExpression(EQUAL, a1.toSymbolReference(), c1.toSymbolReference()),
+                        new ComparisonExpression(EQUAL, b1.toSymbolReference(), e1.toSymbolReference())),
+                ImmutableList.of(a1, b1, c1, d1, d2, e1, e2));
+        assertMultijoinEquals(toMultiJoinNode(joinNode, noLookup(), 3), expected);
+    }
+
+    private static void assertMultijoinEquals(MultiJoinNode actual, MultiJoinNode expected)
+    {
+        assertEquals(ImmutableSet.copyOf(actual.getSources()), ImmutableSet.copyOf(expected.getSources()));
+        assertEquals(ImmutableSet.copyOf(extractConjuncts(actual.getFilter())), ImmutableSet.copyOf(extractConjuncts(expected.getFilter())));
```

Comment:
I would move it to the `MultiJoinNode#equals()`. Than you can simply use `assertThat(actualMultiJoinNode).isEqualTo(expectedMultiJoinNode)`